Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -115,8 +117,54 @@ void MappingButton::Detect()
     // Avoid that the button press itself is registered as an event
     Common::SleepCurrentThread(100);
 
-    const auto expr = MappingCommon::DetectExpression(
-        m_reference, dev.get(), m_parent->GetController()->GetDefaultDevice());
+    std::vector<std::future<QPair<QString, QString>>> futures;
+
+    QString expr;
+
+    if (m_parent->GetParent()->IsMappingAllDevices())
+    {
+      for (const std::string& device_str : g_controller_interface.GetAllDeviceStrings())
+      {
+        ciface::Core::DeviceQualifier devq;
+        devq.FromString(device_str);
+
+        auto adev = g_controller_interface.FindDevice(devq);
+
+        auto future = std::async(std::launch::async, [this, devq, adev, device_str] {
+          return QPair<QString, QString>(
```

Comment:
Note that if you use `std::pair`, you can use `std::make_pair` instead which doesn't require hardcoding in the types.